router/jwt.go

```diff
@@ -0,0 +1,29 @@
+package router
+
+import (
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/dgrijalva/jwt-go/request"
+	"github.com/gin-gonic/gin"
+)
+
+// Auth authenticates request validating JWT signature by a given public key.
+// a user agent must provide a BEARER token via the Authorization header
+func Auth() gin.HandlerFunc {
+	publicKeyPEM, _ := ioutil.ReadFile(os.Getenv("JWT_PUBLIC_KEY_PATH"))
```

Comment:
I would prefer these to be strings read directly from the env.  We typically deploy via Heroku and Docker, injecting configuration entirely via the env, and making no assumptions about file access.